pkg/logger/logger_test.go

```diff
@@ -0,0 +1,246 @@
+package logger
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"knative.dev/pkg/logging"
+)
+
+// Utility functions copied from main.go
+// These is the graceful shutdown logic we want to test
+type CustomHandler struct {
+	wg      *sync.WaitGroup
+	handler http.Handler
+}
+
+func NewCustomHandler(wg *sync.WaitGroup, handler http.Handler) *CustomHandler {
+	return &CustomHandler{wg: wg, handler: handler}
+}
+
+func (h *CustomHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	h.wg.Add(1)
+	defer h.wg.Done()
+	h.handler.ServeHTTP(w, r)
+}
+
+// simple handler that logs the request
+// and sleeps for 5 seconds
+// to simulate a long running request
+func makeLoggingHandler(ctx context.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		logger := logging.FromContext(ctx)
+		select {
+		case <-time.After(5 * time.Second):
+			// simulate a long running request, in case ctx.Done() (srv.ShutdownServer)
+			// is not called fast enough
+		case <-ctx.Done():
+			break
+			// the context will be canceled when the server is shutting down
+			// this gives an exact timing of workload simulation
+		}
```

Comment:
this will exit the select after at most 5 seconds, but also when `ctx.Done()` is triggered. I propose that we make `ctx.Done()` the first statement (no need to `break`) & the `time.After()` second. Timing out these requests after `1s` seems a better lower value (`5s` is long time to wait in tests).